src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEquals.cs

```diff
@@ -173,87 +61,77 @@ internal override void GenerateMethodBody(TypeCompilationState compilationState,
+                    //     other != null &&
+                    //     EqualityContract == other.EqualityContract &&
+                    //     field1 == other.field1 && ... && fieldN == other.fieldN;
+
+                    // other != null
+                    Debug.Assert(!other.Type.IsStructType());
+                    retExpr = F.ObjectNotEqual(other, F.Null(F.SpecialType(SpecialType.System_Object)));
+
+                    // EqualityContract == other.EqualityContract
+                    var contractsEqual = F.Call(receiver: null, F.WellKnownMethod(WellKnownMember.System_Type__op_Equality),
+                                                F.Property(F.This(), _equalityContract),
+                                                F.Property(other, _equalityContract));
+
+                    retExpr = retExpr is null ? (BoundExpression)contractsEqual : F.LogicalAnd(retExpr, contractsEqual);
                 }
                 else
                 {
-                    // This method is the strongly-typed Equals method where the parameter type is
-                    // the containing type.
+                    MethodSymbol? baseEquals = ContainingType.GetMembersUnordered().OfType<SynthesizedRecordBaseEquals>().Single().OverriddenMethod;
 
-                    if (_otherEqualsMethod is null)
+                    if (baseEquals is null || !baseEquals.ContainingType.Equals(ContainingType.BaseTypeNoUseSiteDiagnostics, TypeCompareKind.AllIgnoreOptions) ||
+                        baseEquals.ReturnType.SpecialType != SpecialType.System_Boolean)
                     {
-                        // There are no base record types.
-                        // The definition of the method is as follows (see A.Equals(A) above):
-                        //
-                        // virtual bool Equals(T other) =>
-                        //     other != null &&
-                        //     EqualityContract == other.EqualityContract &&
-                        //     field1 == other.field1 && ... && fieldN == other.fieldN;
-
-                        // other != null
-                        Debug.Assert(!other.Type.IsStructType());
-                        retExpr = F.ObjectNotEqual(other, F.Null(F.SpecialType(SpecialType.System_Object)));
-
-                        // EqualityContract == other.EqualityContract
-                        var contractsEqual = F.Call(receiver: null, F.WellKnownMethod(WellKnownMember.System_Type__op_Equality),
-                                                    F.Property(F.This(), _equalityContract),
-                                                    F.Property(other, _equalityContract));
-
-                        retExpr = retExpr is null ? (BoundExpression)contractsEqual : F.LogicalAnd(retExpr, contractsEqual);
+                        // There was a problem with overriding of base equals, an error was reported elsewhere
+                        F.CloseMethod(F.ThrowNull());
+                        return;
```

Comment:
Same comment here. Can we avoid generating in such error scenarios?

Reply:
> Same comment here. Can we avoid generating in such error scenarios?

Same response. It might or might not work. I'd rather not spend time on this experiment, unless you see specific value in that. 

---
In reply to: [451908196](https://github.com/dotnet/roslyn/pull/45647#discussion_r451908196) [](ancestors = 451908196)